lib/cfn-nag/util/string_to_boolean.rb

```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+# Migrated from multiple classes with no modifications.
+# I, @byronic, worry that this function isn't _quite_ what it says on the tin
+def to_boolean(string)
```

Comment:
yea this name is misleading per your worry.  this code is sort of a "weak" typing for "true", "True", "False", "false", true, false to be treated the same in comparison.

i'm not sure what the best name would be though... it could be as simple as is_true?().  you could indicate something in the name about how it handles string and boolean... that name might be pretty awkward: is_true_weak_typing?() lol.

another tact is to possibly name it after the context .... is_cfn_true? since cfn is the one treating all these string/boolean values the same when it comes to time to evaluate the resource

Reply:
Currently this also treats any non-truthy value (e.g. 'blue') as false. Is that the expected behavior?

If so, I'm cool with `is_true` or `is_truthy` as a renaming scheme.

Reply:
(was clearing out my inbox)

`to_boolean` is not misleading - it attempts to convert a string value to a Boolean, just in a fairly naive manner ("blue" should raise a maybe an ArgumentError, and the method should probably handle "0"/"1", "yes", "no"). There's a good deal of prior art using this method name, and of course Ruby classes are full of `to_` methods.

An OO way to handle this that addresses Eric's note about context is to create a CfnString subclass of string and define a `.to_boolean` method that attempts to replicate CFN's rules for truthy string booleans.